### PR TITLE
treewide: Initial role commit

### PR DESCRIPTION
**Pull request by haslersn**
TODOs:

- [x] `README.md`

This will be used in order to deploy our RFID door.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+That binary then only needs to be installed once (i.e. by one crate). 
+
+### Crate Objects
+
+A crate object is a dictionary which can contain the following keys.
+
+| Key                  |  Default   | Description                                         |
+| :------------------- | :--------: | :-------------------------------------------------- |
+| `archive_url`        | _optional_ | See [source specification](#source-specification)   |
+| `binay_url`          | _optional_ | See [source specification](#source-specification)   |
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
| `binary_url`         | _optional_ | See [source specification](#source-specification)   |
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+One way to obtain that checksum is to insert a fake checksum (for example the one above), run
+Ansible and take the actual checksum from the Ansible error.
+The reason why `checksum` is _required_ is to enforce purity, i.e. the same rust-deploy
+configuration always yields the same installed software.
+
+### Source Specification
+
+There are multiple ways to install a crate.
+Chosing among them is done by providing at exactly one of the following keys per
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
Chosing among them is done by providing exactly one of the following keys per
```

File: handlers/main.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    daemon_reload: True
+
+- name: Actually restart services
+  service:
+    name: "{{ item }}"
+    enabled: True
+    force: True
+    state: restarted
+  loop: "{{ rust_deploy_services.keys() | list }}"
+  listen: Restart services
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]: No newline at end of file

File: tasks/build.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+---
+# This task file assumes that {{ item.key }} is the name and {{ item.value }}
```
Comment (SlothOfAnarchy) [inline tasks/build.yml:2]: For future reference: You can change the variable name of the looped item using [loop_control](https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html#loop-control)

File: handlers/main.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+---
+- name: Restart services
```
Comment (SlothOfAnarchy) [inline handlers/main.yml]: Running this handler is only necessary when changing the systemd service file.
I suggest renaming this handler to `Reload systemd configuration` and calling it only on that occasion. Ansible's `notify` may also be a list in order to call multiple handlers.

Also, `Actually restart services` is never directly invoced, but every time `Restart services` runs, so you could just add `    daemon_reload: True` to the `Actually restart services` handler and achieve the same outcome as now.

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: handlers/main.yml (hunk 5 above)
Comment (haslersn) [reply handlers/main.yml]: That would reload the daemon multiple times.

Comment (haslersn) [reply handlers/main.yml]: Done

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**